app.py

```diff
@@ -0,0 +1,39 @@
+from flask import (
+    jsonify,
+    render_template,
+    session,
+    Flask,
+)
+from json import loads
+
+from game import Game
+
+
+app = Flask(__name__)
+app.secret_key = 'super secret key'
+app.game = Game()
+
+
+@app.route('/')
+def index():
+    json = loads('{"min": 3, "max": 12, "piles": 3}')
+    session['state'] = app.game.new_game(
+        min=json['min'],
+        max=json['max'],
+        piles=json['piles'],
+    )
+    return render_template(
+        'index.html',
+        state=session['state']
+    )
+
+
+@app.route('/update')
+def update_game_state():
+    new_state = session['state']  # minus player move from json
+    session['state'] = app.game.random.move(new_state)
```

Comment:
For here something like this:
```
    move = app.game.random.move(new_state)
    app.game.update(new_state, move)
```

And in `Game`, you'd have:
```
    def update(self, state, move):
        if not self.is_valid_move(state, move):
            raise NimException("Cheater!")
        pile, stones = move
        state[pile] -= stones
```

Reply:
I see. Then the bots won't be generating the new state. That's much more DRY. I like it.